tests/foreman/ui/test_ansible.py

```diff
@@ -448,6 +448,50 @@ def test_positive_assign_and_remove_ansible_role_to_host(self, target_sat, funct
                 == 'No roles assigned directly to the host'
             )
 
+    @pytest.mark.tier2
+    def test_positive_assign_and_remove_role_to_hostgroup(
+        self, target_sat, module_org, module_location
+    ):
+        """Add and remove functionality for ansible roles in hostgroup
+
+        :id: 5d94a484-92c1-4387-ab92-0649e4c4f907
+
+        :steps:
+            1. Import all roles available by default.
+            2. Create a hostgroup
+            3. Assign hostgroup to the host
+            4. Navigate to the hostgroup page.
+            5. Add all ansible roles to the hostgroup
+            6. Remove all added ansible roles from the hostgroup
+
+        :expectedresults: The Ansible Role is successfully assign/add/remove from the hostgroup, and no longer shows
+            up on the UI
+        """
+        hostgroup = target_sat.api.HostGroup(
+            location=[module_location],
+            organization=[module_org],
+        ).create()
+        host = target_sat.api.Host(
+            organization=module_org, location=module_location, hostgroup=hostgroup
+        ).create()
+        hstgrp_name = host.hostgroup.read().name
+        with target_sat.ui_session() as session:
+            synced_all_role = session.ansibleroles.import_all_roles()
+            total_imported_roles = session.ansibleroles.imported_roles_count
+            # verify all roles are synced
+            assert synced_all_role == total_imported_roles
+
+            session.location.select(module_location.name)
+            session.organization.select(module_org.name)
```

Comment:
I strategically included both lines later in the UI session. If i add it start of the ui session it will change the location + org but during role import, the host wasn't visible, making it impossible to import roles for the host.

If the role has been imported to the host at start, it becomes accessible across various locations and organizations.